app/models/host/base.rb

```diff
@@ -372,6 +372,25 @@ def uptime_seconds
 
     private
 
+    def parse_ip_address(address, ignore_link_local: true)
+      # This works around https://bugs.ruby-lang.org/issues/8464 and strips off
+      # the interface identifier if present
+      return nil unless address
+
+      begin
+        addr = IPAddr.new(address)
+      rescue IPAddr::InvalidAddressError
+        if address.is_a?(String) && address.include?('%')
+          addr = IPAddr.new(address.split('%').first)
+        else
+          logger.warn "Ignoring invalid IP address #{address}"
+          return nil
+        end
+      end
+
+      (ignore_link_local && addr.link_local?) ? nil : addr.to_s
```

Comment:
`link_local?` is not available in all the ruby versions we support, that's why tests are failing.

Basically the method does this:
> Returns true if the ipaddr is a link-local address. IPv4 addresses in 169.254.0.0/16 reserved by RFC 3927 and Link-Local IPv6 Unicast Addresses in fe80::/10 reserved by RFC 4291 are considered link-local.

So you can replace it by `IPAddr.new('fe80::/10').include?(ip)` for IPv6 and `IPAddr.new('169.254.0.0/16').include?(ip)`.

```ruby
if ignore_link_local
  return if addr.ipv4? && IPAddr.new('169.254.0.0/16').include?(addr)
  return if addr.ipv6? && IPAddr.new('fe80::/10').include?(addr)
end

addr.to_s
```

Reply:
Of course, it was too easy. Looks like Ruby 2.5 introduced this so I'll update it.

Reply:
Updated to first attempt the modern method. Once we drop Ruby 2.3 (1.25 I think) we can drop the else part.